### PR TITLE
ADMISSION_SYNTAX integration into public X509v3 header

### PR DESCRIPTION
**Pull request by matthiasballreich**
Integration of the ADMISSION_SYNTAX Extension which was added last week into the public X509v3 Header. Issue was opened here: https://github.com/openssl/openssl/issues/3780

So i added the header stuff from v3_admins.h into the include/openssl/x509v3.h and removed the v3_admins.h vom crypto/x509v3 folder. I also updated the util/libcrypto.num with the needed functions, e.g. ADMISSION_SYNTAX_free.

Compilation on Windows x86 was done without problems.

### CODE REVIEW

**Comment by matthiasballreich** [comment]
@richsalz for me it is trivial, but if not what have i to do for the CLA?

**Comment by richsalz** [comment]
See https://www.openssl.org/policies/cla.html  Thanks!

File: include/openssl/x509v3.h [hunk 1]
```diff
@@ -862,8 +862,50 @@ int X509v3_addr_validate_resource_set(STACK_OF(X509) *chain,
...
+
+DEFINE_STACK_OF(ASN1_STRING)
+DEFINE_STACK_OF(PROFESSION_INFO)
+DEFINE_STACK_OF(ADMISSIONS)  
 
 # ifdef  __cplusplus
 }
 # endif
-#endif
+#endif
```
Comment (FdaSilvaYY) [inline include/openssl/x509v3.h]: Your second push add this useless change at the end of this file.

File: include/openssl/x509v3.h [hunk 2]
```diff
@@ -862,8 +862,50 @@ int X509v3_addr_validate_resource_set(STACK_OF(X509) *chain,
...
+DECLARE_ASN1_ITEM(ADMISSIONS)
+
+typedef struct AdmissionSyntax_st {
+    GENERAL_NAME* admissionAuthority;
+    STACK_OF(ADMISSIONS)* contentsOfAdmissions;
+} ADMISSION_SYNTAX;
+
+DECLARE_ASN1_ITEM(ADMISSION_SYNTAX)
+
+DECLARE_ASN1_FUNCTIONS(NAMING_AUTHORITY)
```
Comment (FdaSilvaYY) [inline include/openssl/x509v3.h]: May be you could try to use `DECLARE_ASN1_FUNCTIONS_const` instead of `DECLARE_ASN1_FUNCTIONS`

Comment (matthiasballreich) [inline include/openssl/x509v3.h]: what is the difference between `DECLARE_ASN1_FUNCTIONS_const` and `DECLARE_ASN1_FUNCTIONS`?

I don't have many knowledge in coding internal openssl stuff. I only want to fix the problem that the data structures where not in the public headers. And this worked with my small changes (copy the content of your header (v3_admins.h) into the openssl publich eader (x509v3.h).

File: include/openssl/x509v3.h (hunk 1 above)
Comment (matthiasballreich) [inline include/openssl/x509v3.h]: fixed it. was a small mistake by me :/

**Comment by levitte** [comment]
Re trivial, it might feel trivial in action, but we need to deal with the "trivial" concept from a legal standpoint, and I can't see this as trivial in that regard.

**Comment by matthiasballreich** [comment]
@levitte ok. Yesterday I sent the CLA by email.

**Comment by levitte** [comment]
It hasn't arrived yet...  not to worry, you don't need to resend, it's been caught in a greylist filter.  I need to do some whitelisting.

**Comment by levitte** [comment]
Done.  We should get your email shortly...

**Comment by levitte** [comment]
Another matter is that you need to rebase and remake `util/libcrypto.num`

**Comment by matthiasballreich** [comment]
yes i saw it, you have done some changes on `util/libcrypto.num`. I will fix this later.
With rebase did you mean to call `git rebase master` on my local repository?

**Comment by richsalz** [comment]
There really was no creative thought involved, it’s just putting one file into another.  Files we wrote.

**Comment by richsalz** [comment]
Yes do this:
```
% git rebase master
% git push -f github HEAD
```

**Comment by matthiasballreich** [comment]
`util/libcrypto.num` fixed

**Pull request closed without merging**